jwk/helper.go

```diff
@@ -44,12 +44,16 @@ func EnsureAsymmetricKeypairExists(ctx context.Context, r InternalRegistry, alg,
 }
 
 func GetOrGenerateKeys(ctx context.Context, r InternalRegistry, m Manager, set, kid, alg string) (private *jose.JSONWebKey, err error) {
-	getLock(set).Lock()
-	defer getLock(set).Unlock()
-
+	getLock(set).RLock()
```

Comment:
I'm wondering why the lock is there in the first place. We're using a SQL backend for this, which is atomic.

Reply:
that's a good point..

Reply:
@terev if it were to keep the `map[string]*RWMutex` or maybe just `map[string]*Mutex` and lock on key generation, do you think that would work for the key set initialization condition?

Reply:
Yeah I think a plain mutex per keyset is fine. But maybe keep the rwlock for accessing the map. But actually use the read lock in the right case there. But to be clear I'm not part of ory so it aesnar disagrees that's cool too.

Reply:
> But to be clear I'm not part of ory so if aeneasr disagrees that's cool too.
thanks anyways @terev 😄!

Okay, @aeneasr, I think I'm following what you're saying about not even needing the read lock.

I'll read up on the `test/conformance` to grok what's going on there 🤞🏻

Also found a couple snapshots that looked like they needed a quick patch so I grabbed those just in case.